core/qthelper.cpp

```diff
@@ -984,6 +984,28 @@ QString thumbnailFileName(const QString &filename)
 	return thumbnailDir() + hash.result().toHex();
 }
 
+int video_duration(const QString &filename)
```

Comment:
Sorry, but I really dislike the fact that the thumbnailer internals are reproduced here. I'd rather either use a thumbnailer function to fetch the thumbnail or extend the `PictureEntry` with a duration field. I think I would prefer the second option. I'll look into it when I find some time...

Reply:
My proposal would be to extend the `PictureEntry` struct by a `duration` field, which is initialized to 0. Then, fill this field in `DivePictureModel::updateThumbnail()` and access it like the offset with a `Qt::UserRole + X`.

This would mean the user has to wait until the thumbnails have loaded, but I think that's acceptable.